src/sync/impl/sync_file.cpp

```diff
@@ -248,27 +250,39 @@ std::string SyncFileManager::get_base_sync_directory() const
     return sync_path;
 }
 
-std::string SyncFileManager::user_directory(const std::string& user_identity) const
+std::string SyncFileManager::user_directory(const std::string& local_identity,
+                                            util::Optional<SyncUserIdentifier> user_info) const
 {
-    REALM_ASSERT(user_identity.length() > 0);
-    if (filename_is_reserved(user_identity)) {
+    REALM_ASSERT(local_identity.length() > 0);
+    if (filename_is_reserved(local_identity))
         throw std::invalid_argument("A user can't have an identifier reserved by the filesystem.");
-    }
+
     auto user_path = file_path_by_appending_component(get_base_sync_directory(),
-                                                      user_identity,
+                                                      local_identity,
                                                       util::FilePathType::Directory);
-    util::try_make_dir(user_path);
+    bool success = util::try_make_dir(user_path);
+
+    if (success && user_info && user_info->user_id != local_identity) {
+        // Add a text file in the user directory containing the user identity, for backup purposes.
```

Comment:
It might be worth stating that we're intentionally ignoring errors (if it is in fact intentional that we're doing so).